src/casts/specify-representation.rkt

```diff
@@ -143,541 +156,339 @@ but a static single assignment is implicitly maintained.
            (sr-alloc "dynamic_boxed" DYN-BOXED-TAG
                      `(("" . ,val-var) ("" . ,type-var)))))]))
   
-  (: get-mk-tuple-type! (Uid -> (Code-Label Uid)))
-  (define (get-mk-tuple-type! mk-glb)
+  (: get-mk-tuple-type-glb! (Uid -> (Code-Label Uid)))
+  (define (get-mk-tuple-type-glb! mk-tglb)
     (: make-code! (Uid -> (Code-Label Uid)))
-    (define (make-code! mk-glb)
-      (define mk-tuple-u (next-uid! "make-tuple-type"))
-      (define t1-u    (next-uid! "tuple-type1"))
-      (define t2-u    (next-uid! "tuple-type2"))
-      (define i-u     (next-uid! "index"))
-      (define a-u     (next-uid! "tuple-type-num"))
-      (define t1a-u   (next-uid! "tuple-type1-argument"))
-      (define t2a-u   (next-uid! "tuple-type2-argument"))
-      (define c1-u    (next-uid! "resulting_type"))
-      (define c2-u    (next-uid! "resulting_type"))
-      (define ca-u    (next-uid! "argument-type"))
-      (define t1      (Var t1-u))
-      (define t2      (Var t2-u))
-      (define c1      (Var c1-u))
-      (define c2      (Var c2-u))
-      (define i       (Var i-u))
-      (define a       (Var a-u))
-      (define mk-tuple   (Code-Label mk-tuple-u))
-      (define mk-g    (Code-Label mk-glb))
-      (define mk-tuple-t : D0-Code
-        (Code `(,t1-u ,t2-u ,i-u ,a-u)
-          (If (Op '= `(,i ,a))
+    (define (make-code! mk-tglb)
+      (define-vars mk-tuple-type-glb t1 t2 i count t1a t2a t tmp-t t-glb)
+      (define mk-tuple-type-glb-label (Code-Label mk-tuple-type-glb))
+      (define mk-tglb-label (Code-Label mk-tglb))
+      (define mk-tuple-type-glb-c : D0-Code
+        (Code `(,t1 ,t2 ,i ,count)
```

Comment:
This code could be further simplified by use of the code$ macro.

Reply:
I will implement it using `code$` macro when I move it from `specify-representation`.